chalice/watcher/stat.py

```diff
@@ -0,0 +1,66 @@
+import threading
+import time
+
+from typing import Dict, List, Set, Callable  # noqa
+
+from chalice.watcher.shared import Watcher
+from chalice.utils import OSUtils
+
+
+class StatFileObserver(object):
```

Comment:
Trying this out, there's several parity issues I noticed between this and watchdog:

* It doesn't trigger reloads on deletes of files
* Watchdog has some sort of configuration for dealing with hidden files.  I noticed this because when I opened a file with vim and started typing, a reload would be triggered even though I didn't save the file.  This appears to be because of the .swp file that vim creates.  For whatever reason, watchdog handles this somehow and only refreshes the file when I actually save the file I've opened, which is the behavior I'd expect.